src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresDatatypeIT.java

```diff
@@ -102,18 +194,29 @@ public void testWriteToTableWithUuidColumn() throws SQLException {
         .put("lastname", "Brams")
         .put("jsonid", jsonid)
         .put("userid", uuid.toString());
-    task.put(Collections.singleton(new SinkRecord(tableName, 1, null, null, schema, struct, 1)));
-    try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
-      try (Statement s = c.createStatement()) {
-        try (ResultSet rs = s.executeQuery("SELECT * FROM " + tableName)) {
-          assertTrue(rs.next());
-          assertEquals(struct.getString("firstname"), rs.getString("firstname"));
-          assertEquals(struct.getString("lastname"), rs.getString("lastname"));
-          assertEquals(struct.getString("jsonid"), rs.getString("jsonid"));
-          assertEquals(struct.getString("userid"), rs.getString("userid"));
-        }
-      }
-    }
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(tableName, schema, struct));
+    connect.kafka().produce(tableName, null, kafkaValue);
+
+    waitForCondition(
```

Comment:
If we end up using the wait-for-offset-commit strategy proposed above, we can do that here and leave the database content assertions out of the `waitForCondition` call.

Reply:
I went with a different approach, so this might not work.